src/input/ime.c

```diff
@@ -304,9 +304,7 @@ handle_keyboard_grab_destroy(struct wl_listener *listener, void *data)
 {
 	struct input_method_relay *relay =
 		wl_container_of(listener, relay, keyboard_grab_destroy);
-	struct wlr_input_method_keyboard_grab_v2 *keyboard_grab =
-		relay->input_method->keyboard_grab;
-
+	struct wlr_input_method_keyboard_grab_v2 *keyboard_grab = data;
```

Comment:
Suggest adding an `assert(data)` or `assert(keyboard_grab)` here to ensure that we remember to handle this when it is changed in future wlroots releases.